tests/bluetooth/bsim_bt/bsim_test_eatt/src/main_collision.c

```diff
@@ -0,0 +1,221 @@
+
+	default_conn = bt_conn_ref(conn);
+	printk("Connected: %s\n", addr);
+	is_connected = true;
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Disconnected: %s (reason 0x%02x)\n", addr, reason);
+
+	if (default_conn != conn) {
+		FAIL("Conn mismatch disconnect %s %s)\n", default_conn, conn);
+	}
+
+	bt_conn_unref(default_conn);
+	default_conn = NULL;
+	is_connected = false;
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+static void test_peripheral_main(void)
+{
+	int err;
+	uint16_t num_enhanced;
+
+	err = bt_enable(NULL);
+	if (err) {
+		FAIL("Can't enable Bluetooth (err %d)\n", err);
+	}
+
+	err = bt_le_adv_start(BT_LE_ADV_CONN_NAME, ad, ARRAY_SIZE(ad), NULL, 0);
+	if (err) {
+		FAIL("Advertising failed to start (err %d)\n", err);
+	}
+
+	while (!is_connected) {
+		k_sleep(K_MSEC(100));
+	}
+
+	err = bt_eatt_connect(default_conn, CONFIG_BT_EATT_MAX);
+	if (err) {
+		FAIL("Sending credit based connection request failed (err %d)\n", err);
+	}
+
+	/* Wait for EATT channels to be connected */
+	k_sleep(K_MSEC(10000));
+	num_enhanced = bt_att_num_enhanced_bearers(default_conn);
+	printk("%d enhanced bearers connected\n", num_enhanced);
+	if (num_enhanced != CONFIG_BT_EATT_MAX) {
+		FAIL("Expected %d enhanced bearers, got %d\n", CONFIG_BT_EATT_MAX, num_enhanced);
+	}
+	/* Disconnect */
+	err = bt_conn_disconnect(default_conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+	if (err) {
+		FAIL("Disconnection failed (err %d)\n", err);
+	}
+
+	while (is_connected) {
+		k_sleep(K_MSEC(100));
+	}
+
+	PASS("EATT Peripheral tests Passed\n");
+}
+
+static void device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
+			 struct net_buf_simple *ad)
+{
+	struct bt_le_conn_param *param;
+	int err;
+
+	err = bt_le_scan_stop();
+	if (err) {
+		FAIL("Stop LE scan failed (err %d)\n", err);
+	}
+
+	param = BT_LE_CONN_PARAM_DEFAULT;
+	err = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN, param, &default_conn);
+
+	if (err) {
+		FAIL("Create conn failed (err %d)\n", err);
+	}
+	printk("Device connected\n");
+}
+
+static void test_central_main(void)
+{
+	struct bt_le_scan_param scan_param = {
+		.type = BT_LE_SCAN_TYPE_ACTIVE,
+		.options = BT_LE_SCAN_OPT_NONE,
+		.interval = BT_GAP_SCAN_FAST_INTERVAL,
+		.window = BT_GAP_SCAN_FAST_WINDOW,
+	};
+
+	int err;
+	uint16_t num_enhanced;
+
+	err = bt_enable(NULL);
+	if (err) {
+		FAIL("Can't enable Bluetooth (err %d)\n", err);
+	}
+
+	err = bt_le_scan_start(&scan_param, device_found);
+	if (err) {
+		FAIL("Scanning failed to start (err %d)\n", err);
+	}
+
+	while (!is_connected) {
+		k_sleep(K_MSEC(100));
+	}
+
+	err = bt_eatt_connect(default_conn, CONFIG_BT_EATT_MAX);
+	if (err) {
+		FAIL("Sending credit based connection request failed (err %d)\n", err);
+	}
+
+	/* Wait for EATT channels to be connected */
+	k_sleep(K_MSEC(10000));
```

Comment:
Better to wait for flag set by callback

Reply:
There is currently no callback to the application when ATT channels are connected. I am planning another PR which makes it possible for the application to connect EATT channels (potentially automatically). A callback can be added then.

Reply:
I get that there are no callback for ATT, but EATT is dynamic channels, right? Wouldn't the `connected` callback in `bt_l2cap_chan_ops` be called? 

But if we are adding this in a separate PR, please add a TODO here.

Reply:
`ops->connected` is `bt_att_connected()`, so there is a callback into the ATT layer, but not to the application. I am thinking we can add a callback similar to `att_mut_updated` that exists in `gatt.c/bt_gatt_cb`.

Reply:
Typically a GATT user would not need to care about EATT connections. Perhaps we need an ATT callback instead?

Reply:
Yes, that makes more sense.

When making the api for connecting EATT channels available to the application it needs a way to know that channels are connected. I think the callback should be added in that PR instead and leave the sleep here for now. Will add a TODO.

Reply:
An API question for that PR: Would it be preferable to have one callback per channel created or per request?

Reply:
Hmm, given that the stack now will retry requests (possibly with less channels), I think a callback per channel is better, don't you think?

Reply:
I am not sure.

I have been thinking that we can let `bt_eatt_connect()` take any number of channels to connect and then potentially send multiple requests. In that case per-channel callbacks make more sense. The application does not really care about the on-air requests.

Reply:
> The application does not really care about the on-air requests.

Agree.